src/client/Microsoft.Identity.Client/Extensibility/CdtAuthenticationScheme.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.ApiConfig;
+using Microsoft.Identity.Client.AuthScheme;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+#if SUPPORTS_SYSTEM_TEXT_JSON
+using JObject = System.Text.Json.Nodes.JsonObject;
+using JToken = System.Text.Json.Nodes.JsonNode;
+#else
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Json.Linq;
+#endif
+
+namespace Microsoft.Identity.Client
+{
+    //Temporary location
+    public class CdtAuthenticationScheme : IAuthenticationScheme
+    {
+        //CDT
+        public const string CdtKey = "xms_ds_cnf";
+        public const string CdtNonce = "xms_ds_nonce";
+        public const string CdtEncKey = "xms_ds_enc";
+        public const string NoAlgorythmPrefix = "none";
+        public const string JasonWebTokenType = "JWT";
+        public const string CdtTokenType = "CDT";
+        public const string CdtEncryptedAlgoryth = "dir";
+        public const string CdtEncryptedValue = "A256CBC-HS256";
+        public const string CdtRequestConfirmation = "req_ds_cnf";
+
+        private readonly CdtCryptoProvider _cdtCryptoProvider;
+        private readonly string _constraints;
+        private readonly string _dsReqCnf;
+
+        /// <summary>
+        /// Creates Cdt tokens, i.e. tokens that are bound to an HTTP request and are digitally signed.
+        /// </summary>
+        /// <remarks>
+        /// Currently the signing credential algorithm is hard-coded to RSA with SHA256. Extensibility should be done
+        /// by integrating Wilson's SigningCredentials
+        /// </remarks>
+        public CdtAuthenticationScheme(string constraints, X509Certificate2 certificate)
+        {
+            _constraints = constraints ?? throw new ArgumentNullException(nameof(constraints));
+
+            _cdtCryptoProvider = new CdtCryptoProvider(certificate);
+
+            _dsReqCnf = _cdtCryptoProvider.CannonicalPublicKeyJwk;
+        }
+
+        public TokenType TelemetryTokenType => TokenType.Bearer;
+
+        public string AuthorizationHeaderPrefix => Constants.BearerAuthHeaderPrefix;
```

Comment:
The constants need to be public if they must be used in add-ins. Same w/line 66.